File: GP Disktop Application/GestureRecognition/NeuralMatrix.cpp
#include<iostream>
#include <vector>
using namespace std;
class NeuralMatrix
{

public:
    vector<vector<double> >matrix;
    NeuralMatrix(int row,int column)
    {
        matrix.resize(row);
		cout << matrix.size() << endl;
			
           for(int i=0; i<row; i++)
             matrix[i].resize(column);


    }
     void SetMatrixDim(int row,int column)
    {
        matrix.resize(row);
        for(int i=0; i<row; i++)
            matrix[i].resize(column);


    }
// matrix multiplicatiion
    NeuralMatrix  operator*(NeuralMatrix mat)
    {

//if(this->matrix[0].size()==vec.size())
        NeuralMatrix newMat(this->matrix.size(),mat.matrix[0].size());// 1 constant because the column size of input 1 forever

        for(int i=0; i<newMat.matrix.size(); i++)
        {
            for(int j=0; j<newMat.matrix[i].size(); j++)
            {
                newMat.matrix[i][j]=0;
                for(int k=0; k<this->matrix[i].size(); k++)
                {
                    newMat.matrix[i][j]+=this->matrix[i][k]*mat.matrix[k][j];
                }


            }
        }


        return newMat;
    }
    NeuralMatrix  operator+(NeuralMatrix mat)
    {
        for(int i=0; i<this->matrix.size(); i++)
        {
            for(int j=0; j<this->matrix[i].size(); j++)
            {

                this->matrix[i][j]+=mat.matrix[i][j];
            }
        }


        return *this;
    }
};

File: GP Android Application/src/Classifer/NeuralMatrix.java
package Classifer;

import java.util.Vector;

/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
/**
 *
 * @author MohamedAtta
 */
public class NeuralMatrix {
    //  Vector<Vector<Float> >matrix;
  
      float [][] matrix;
      
      
      public  NeuralMatrix(int row,int column)
       {
    	  matrix = new float[row][column];
         /*  matrix=new Vector();
       matrix.setSize(row);
         for(int i=0; i<row; i++)
         {
             matrix.set(i,new Vector());
             matrix.get(i).setSize(column);

         }*/
         
       }

       
       void SetMatrixDim(int row,int column)
       {
    	   matrix = new float[row][column];
    	   /*
          matrix=new Vector();
       matrix.setSize(row);
         for(int i=0; i<row; i++)
         {
             matrix.set(i,new Vector());
             matrix.get(i).setSize(column);
       
         }*/
       
       }
// matrix multiplicatiion
    NeuralMatrix mul(NeuralMatrix mat1,NeuralMatrix mat2)
    {
    int row = mat1.matrix.length;
    int col = mat2.matrix[0].length;
     NeuralMatrix newMat=new NeuralMatrix(row,col);// 1 constant because the column size of input 1 forever

        for(int i=0; i<row; i++)
        {
            for(int j=0; j<newMat.matrix[i].length; j++)
            {
                newMat.matrix[i][j]=0.0f;
                for(int k=0; k<mat1.matrix[i].length; k++)
                {
                newMat.matrix[i][j] = newMat.matrix[i][j];
                }


            }
        }
return newMat;
    }
    // matrix addition
    NeuralMatrix add(NeuralMatrix mat1,NeuralMatrix mat2)
    {
     NeuralMatrix newMat=new NeuralMatrix(mat1.matrix.length,mat1.matrix[0].length);// 1 constant because the column size of input 1 forever
 for(int i=0; i<mat1.matrix.length; i++)
        {
            for(int j=0; j<mat1.matrix[i].length; j++)
            {

                 newMat.matrix[i][j] = mat1.matrix[i][j] +mat2.matrix[i][j];// setElementAt( mat1.matrix.get(i).get(j)+mat2.matrix.get(i).get(j);
           
            }
        }

        return newMat;
    }
    

}
